lib/src/position.dart

```diff
@@ -213,6 +213,254 @@ abstract class Position<T extends Position<T>> {
+        return null;
+      }
+      if (san.length == 4 && san[1] != '@') {
+        return null;
+      }
+      final Role role;
+      if (san.length == 3) {
+        role = Role.pawn;
+      } else if (san.length == 4) {
+        final parsedRole = Role.fromChar(san[0]);
+        if (parsedRole == null) {
+          return null;
+        }
+        role = parsedRole;
+      } else {
+        return null;
+      }
+      final destination = parseSquare(san.substring(san.length - 2));
+      if (destination == null) {
+        return null;
+      }
+      final move = DropMove(to: destination, role: role);
+      if (!isLegal(move)) {
+        return null;
+      }
+      return move;
+    }
+
+    if (san == 'O-O') {
+      Move? move;
+      if (turn == Side.white) {
+        // Castle the king from e1 to g1
+        move = const NormalMove(from: 4, to: 6);
+      }
+      if (turn == Side.black) {
+        // Castle the king from e8 to g8
+        move = const NormalMove(from: 60, to: 62);
+      }
+      if (!isLegal(move!)) {
+        return null;
+      }
+      return move;
+    }
+    if (san == 'O-O-O') {
+      Move? move;
+      if (turn == Side.white) {
+        // Castle the king from e1 to c1
+        move = const NormalMove(from: 4, to: 2);
+      }
+      if (turn == Side.black) {
+        // Castle the king from e8 to c8
+        move = const NormalMove(from: 60, to: 58);
+      }
+      if (!isLegal(move!)) {
+        return null;
+      }
+      return move;
+    }
+
+    final isPromotion = san.contains('=');
+    final isCapturing = san.contains('x');
+    int? pawnRank;
+    if (oneIndex <= san.codeUnits[0] && san.codeUnits[0] <= eightIndex) {
+      pawnRank = san.codeUnits[0] - oneIndex;
+      san = san.substring(1);
+    }
+    final isPawnMove = aIndex <= san.codeUnits[0] && san.codeUnits[0] <= hIndex;
+
+    if (isPawnMove) {
+      // Every pawn move has a destination (e.g. d4)
+      // Optionally, pawn moves have a promotion
+      // If the move is a capture then it will include the source file
+
+      final colorFilter = board.bySide(turn);
+      final pawnFilter = board.byRole(Role.pawn);
+      SquareSet filter = colorFilter.intersect(pawnFilter);
+      Role? promotionRole;
+
+      // We can look at the first character of any pawn move
+      // in order to determine which file the pawn will be moving
+      // from
+      final sourceFileCharacter = san.codeUnits[0];
+      if (sourceFileCharacter < aIndex || sourceFileCharacter > hIndex) {
+        return null;
+      }
+
+      final sourceFile = sourceFileCharacter - aIndex;
+      final sourceFileFilter = SquareSet.fromFile(sourceFile);
+      filter = filter.intersect(sourceFileFilter);
+
+      if (isCapturing) {
+        // Invalid SAN
+        if (san[1] != 'x') {
+          return null;
+        }
+
+        // Remove the source file character and the capture marker
+        san = san.substring(2);
+      }
+
+      if (isPromotion) {
```

Comment:
Please add some tests that validate promotion parsing.

Also here I think it miss Antichess support. See: https://github.com/niklasf/chessops/blob/f287455e1cc83925c9f759f3c8829e2ca89d243f/src/san.ts#L126

Please also add a test that validates Antichess promotion.